src/solvers/dgmulti/flux_differencing_gauss_sbp.jl

```diff
@@ -25,15 +25,262 @@ function tensor_product_quadrature(element_type::Hex, r1D, w1D)
   return rq, sq, tq, wq
 end
 
-# Todo: DGMulti. Decide if we should add GaussSBP on triangles.
+# type parameters for TensorProductFaceOperator
+abstract type AbstractGaussOperator end
+struct Interpolation <: AbstractGaussOperator end
+struct Projection  <: AbstractGaussOperator end
+
+"""
+  struct TensorProductGaussFaceOperator{Tmat, Ti}
+
+Data for performing tensor product interpolation from volume nodes to face nodes.
+"""
+struct TensorProductGaussFaceOperator{NDIMS, OperatorType <: AbstractGaussOperator, Tmat, Tweights, Tindices}
+  interp_matrix_gauss_to_face_1d::Tmat
+  inv_volume_weights_1d::Tweights
+  face_indices_tensor_product::Tindices
+  nnodes_1d::Int
+  nfaces::Int
+end
+
+function TensorProductGaussFaceOperator(operator::AbstractGaussOperator,
+                                        dg::DGMulti{2, Quad, GaussSBP})
+  rd = dg.basis
+
+  rq1D, wq1D = StartUpDG.gauss_quad(0, 0, polydeg(dg))
+  interp_matrix_gauss_to_face_1d = polynomial_interpolation_matrix(rq1D, [-1; 1])
+
+  nnodes_1d = length(rq1D)
+
+  # Permutation of indices in a tensor product form
+  indices = reshape(1:length(rd.rf), nnodes_1d, rd.Nfaces)
+  face_indices_tensor_product = zeros(Int, 2, nnodes_1d, ndims(rd.elementType))
+  for i in 1:nnodes_1d # loop over nodes in one face
+    face_indices_tensor_product[:, i, 1] .= indices[i, 1:2]
+    face_indices_tensor_product[:, i, 2] .= indices[i, 3:4]
+  end
+
+  T_op = typeof(operator)
+  Tm = typeof(interp_matrix_gauss_to_face_1d)
+  Tw = typeof(rd.wq)
+  Ti = typeof(face_indices_tensor_product)
+  return TensorProductGaussFaceOperator{2, T_op, Tm, Tw, Ti}(interp_matrix_gauss_to_face_1d,
+                                                             inv.(wq1D), face_indices_tensor_product,
+                                                             nnodes_1d, rd.Nfaces)
+end
+
+function TensorProductGaussFaceOperator(operator::AbstractGaussOperator,
+                                        dg::DGMulti{3, Hex, GaussSBP})
+  rd = dg.basis
+
+  rq1D, wq1D = StartUpDG.gauss_quad(0, 0, polydeg(dg))
+  interp_matrix_gauss_to_face_1d = polynomial_interpolation_matrix(rq1D, [-1; 1])
+
+  nnodes_1d = length(rq1D)
+
+  # Permutation of indices in a tensor product form
+  indices = reshape(1:length(rd.rf), nnodes_1d, nnodes_1d, rd.Nfaces)
+  face_indices_tensor_product = zeros(Int, 2, nnodes_1d, nnodes_1d, ndims(rd.elementType))
+  for i in 1:nnodes_1d, j in 1:nnodes_1d # loop over nodes in one face
```

Comment:
```suggestion
  for j in 1:nnodes_1d, i in 1:nnodes_1d # loop over nodes in one face
```
Performance doesn't matter here but I would prefer keeping the order of the loops consistent everywhere.